tests/impl.h

```diff
@@ -1,5 +1,288 @@
 #ifndef SSE2NEONTEST_H
 #define SSE2NEONTEST_H
+#include "common.h"
+#define ENUM(c) it_##c,
+#define STR(c) #c,
+#define CASE(c)                   \
+    case it_##c:                  \
+        ret = test_##c(*this, i); \
+        break;
+#define TYPE_FOREACH(TYPE)            \
```

Comment:
I am not sure if `TYPE` is a meaningful name which refers to intrinsics. @marktwtn, how do you think?

Reply:
How about `INTRINSICS`?

Reply:
I think `INTRIN` is reasonable as an abbreviation.